app/src/main/java/com/engineering/dokkan/MainActivity.java

```diff
@@ -1,14 +1,45 @@
 package com.engineering.dokkan;
 
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
 
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
 
 public class MainActivity extends AppCompatActivity {
-
+    Button button_x,button_y;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        button_x=findViewById(R.id.button);
```

Comment:
extract this out in a separate method 

like initializeViews(), setLisetners() and for adding fragment extract it also in a separate method like navigateToNotification()

so onCreate() will be calling  initializeViews() and setLisetners()
and in onClick will call navigateToNotification()

every method should have a single responsibility